### DISCUSSION TITLE
Share auth session between web app and Chrome extension

### DISCUSSION THREAD

**Discussion by mariusa** [supabase/supabase#862] [Questions]
After a user signs up/in on the web app, a Chrome extension should be able to recognize that user without asking again to login.
This works with Firebase, I think by using `signInWithCustomToken`, which is passed from the web app to chrome extension via `runtime.sendMessage()`
This also requires having the extension domain (chrome-extension://ID) whitelisted in Firebase settings.

Here's a Supabase Chrome extension: https://github.com/rustyzone/Supabase-Chrome-Ext-Auth
How could this recognize the session from users already signed in the web app?

Maybe by getting the JWT token (https://github.com/supabase/supabase/discussions/845) from web app and sending it to extension? But there's no `signInWithCustomToken` method, although it should be possible (eg https://github.com/supabase/supabase/discussions/733 ?)

**Comment by connorlindsey** [comment]
This would be amazing because it seems like Supabase's Google login may not work in Chrome extensions (see #5787)

**Comment by datduyng** [comment]
Any solution for this yet?

**Comment by der-ofenmeister** [comment]
See my answer in #5787. Hope it helps! ✌️

**Comment by arcavid** [comment]
https://supabase.com/docs/guides/auth/social-login/auth-google#using-native-sign-in-for-chrome-extensions

**Comment by Wizzel1** [comment]
@mariusa have you found a satisfying solution?

**Comment by danielvoelk** [comment]
I'm also looking for a solution to that, but couldn't find one so far.
